### ISSUE TITLE
Prebuild with leveldown on Elastic Beanstalk

### ISSUE THREAD

**Issue by fergiemcdowall** [prebuild/prebuild#152]
Prebuild seems to be failing with `leveldown` on Elastic Beanstalk.

Full report: https://github.com/Level/leveldown/issues/326

**Comment by daikiuk** [comment]
Experiencing very same issue. Any workarounds?

**Comment by fergiemcdowall** [comment]
Yes- it seems to be a permissions error, to work around it you need to create a file at `.ebextensions/leveldownWorkaround.config` that looks like this:

```bash
files:
  "/opt/elasticbeanstalk/hooks/appdeploy/pre/45makeDirectoriesWritable.sh":
    mode: "000755"
    owner: root
    group: root
    content: |
      #!/usr/bin/env bash
      chmod 777 -R /tmp/.npm
      chmod 777 -R /tmp/deployment/application/node_modules/leveldown
```
If you now `eb deploy`, then leveldown should build as expected.

-note that this fix is just to "get it to work", and is probably being a little generous with permissions.

**Comment by daikiuk** [comment]
@fergiemcdowall thanks!

**Comment by kiwicopple** [comment]
Another possible solution here:
https://stackoverflow.com/questions/46001516/beanstalk-node-js-deployment-node-gyp-fails-due-to-permission-denied/46001517#46001517

tl;dr:
Add a file `.npmrc` to the application root with the content:

```bash
unsafe-perm=true
```

I was having difficulty with @fergiemcdowall's solution for some reason
